aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -218,8 +220,8 @@ async def task_retrieve_job(node, transport_queue, retrieved_temporary_folder, c
     :type cancellable: :class:`aiida.engine.utils.InterruptableFuture`
     :raises: TransportTaskException if after the maximum number of retries the transport task still excepted
     """
-    if node.get_state() == CalcJobState.PARSING:
-        logger.warning(f'CalcJob<{node.pk}> already marked as PARSING, skipping task_retrieve_job')
+    if node.get_state() == CalcJobState.STASHING:
+        logger.warning(f'CalcJob<{node.pk}> already marked as STASHING, skipping task_retrieve_job')
```

Comment:
should say `task_stash_job`?

Reply:
nope. This code is in the `task_retrieve_job`. Since stashing is done before retrieving, this is a safeguard to ensure it is only executed once.